src/Raven.Server/Documents/CollectionRunner.cs

```diff
@@ -132,12 +133,6 @@ public virtual Task<IOperationResult> ExecuteDelete(string collectionName, Colle
             };
         }
 
-        private static bool IsHiLoDocument(Document document)
-        {
-            var collection = CollectionName.GetCollectionName(document.Data);
```

Comment:
Another option, we could use here `CollectionName.GetLazyCollectionNameFrom` and always compare `LazyStringValues`

Reply:
I was trying to avoid accessing the data since this will fetch the data to memory.
But we can compare the lazy string id